src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -280,23 +287,26 @@ private boolean isLetterHolderRole(String role) {
         }
 
         User letterHolder = userService.getUser(authorisation);
-        return Optional.of(readCase(letterHolder, letterHolderCases.get(0)));
+        CaseDetails caseDetails = readCase(letterHolder, letterHolderCases.get(0));
+        if (CREATE.getValue().equals(caseDetails.getState())) {
```

Comment:
Use the predicate created. No need to do if condition again. if you need negate use predicate.not

Reply:
It's simple if condition and not a method reference or lambda expression. So not a good candidate for functional programming here.

Reply:
Its not to do with Functional Programming. Since you had Predicate already, so we use it. If else, the Predicate you are using becomes unnecessary.

Reply:
Declare once and use multiple times.